### PR TITLE
[build.yaml] eliminate hail_query_gcs_path

### PR DESCRIPTION
**Pull request by danking**
After this deploys and we re-run terraform, I can remove the deprecated entry.

### CODE REVIEW

File: infra/azure/main.tf [hunk 1]
```diff
@@ -306,6 +306,12 @@ resource "azurerm_storage_container" "batch_logs" {
   container_access_type = "private"
 }
 
+resource "azurerm_storage_container" "query" {
+  name                  = "query"
+  storage_account_name  = azurerm_storage_account.batch.name
+  container_access_type = "private"
+}
```
Comment (daniel-goldstein) [inline infra/azure/main.tf:313]: As is, only the batch service principal will have access to this container. Do any other service accounts need access?

File: query/deployment.yaml [hunk 2]
```diff
@@ -96,13 +96,13 @@ spec:
            - name: HAIL_QUERY_WORKER_IMAGE
              value: {{ query_image.image }}
 {% if deploy %}
-           - name: HAIL_QUERY_GCS_PATH
+           - name: HAIL_QUERY_STORAGE_UI
```
Comment (daniel-goldstein) [inline query/deployment.yaml:99]: nit: can move the name out of the `if` to avoid having to write the same name twice.

**Review by daniel-goldstein — CHANGES_REQUESTED** [review]
Looks fine just a question about access control

File: infra/azure/main.tf (hunk 1 above)
Comment (danking) [reply infra/azure/main.tf:313]: No, in the long run Batch workers will fetch the JAR.

Comment (daniel-goldstein) [reply infra/azure/main.tf:313]: In that case you'll need to give the batch worker managed identity access to the container

**Review by daniel-goldstein — DISMISSED** [review]
See comment about batch worker permissions

File: infra/azure/main.tf (hunk 1 above)
Comment (danking) [reply infra/azure/main.tf:313]: This is a new container, not a new account. AFAIK, the batch worker already has access to the batch storage account.

Comment (daniel-goldstein) [reply infra/azure/main.tf:313]: Ah, you're right! Never mind.

**Review by daniel-goldstein — APPROVED** [review]

**Pull request merged**